lib/persisted-model.js

```diff
@@ -619,6 +668,20 @@ module.exports = function(registry) {
       rest: {after: convertNullToNotFoundError}
     });
 
+    setRemoting(PersistedModel, 'replaceById', {
+      description: 'Replace attributes for a model instance and persist it into the data source.',
+      accessType: 'WRITE',
+      accepts: [
+        { arg: 'id', type: 'any', description: 'Model id', required: true,
+          http: { source: 'put' }},
+          { arg: 'data', type: 'object', http: { source: 'body' }, description:
+        'An object of model property name/value pairs' },
+      ],
+      returns: { arg: 'data', type: typeName, root: true },
+      http: { verb: configurableVerb.replaceById, path: '/:id' },
+      rest: { after: convertNullToNotFoundError },
```

Comment:
Could you please check that `replaceById` returns `null` when the `id` was not found? I don't remember if we implemented this feature. I think this may have been the case of `prototype.updateAttributes` because of the way how shared constructor works.

For `replaceById`, I think it makes more sense to raise not-found-error directly from `replaceById` implementation, so that we don't need this extra bit here.